common/bytes.go

```diff
@@ -117,28 +126,29 @@ func CopyBytes(b []byte) (copiedBytes []byte) {
 	return
 }
 
-func HasHexPrefix(str string) bool {
-	l := len(str)
-	return l >= 2 && str[0:2] == "0x"
-}
+// isHexPattern is a regular expression to verify whether a string represents an
+// abirarilly long hex number with or without the "0x" prefix.
+var isHexPattern = regexp.MustCompile("^0(x|X)([0-9a-fA-F]{2})+$")
 
+// IsHex checks whether a string is a valid hexadecimal number, consisting of
+// only hex digits and prefixed with 0x or 0X.
 func IsHex(str string) bool {
-	l := len(str)
-	return l >= 4 && l%2 == 0 && str[0:2] == "0x"
-}
-
-func Bytes2Hex(d []byte) string {
-	return hex.EncodeToString(d)
+	return isHexPattern.MatchString(str)
 }
 
 func Hex2Bytes(str string) []byte {
-	h, _ := hex.DecodeString(str)
-
+	h, err := hex.DecodeString(str)
+	if err != nil {
+		glog.V(logger.Error).Infof("Invalid hex string to decode: %s: %v", str, err)
+	}
```

Comment:
I don't agree with error logging in a primitive conversion function.
If the caller wants to know whether the input is valid, it can use hex.DecodeString
and handle the error.